ios/PolyPodApp/PodApi/PostOffice.swift

```diff
@@ -162,25 +173,39 @@ extension PostOffice {
     }
     
     private func handlePolyInDelete(args: [Any], completionHandler: @escaping (MessagePackValue?, MessagePackValue?) -> Void) {
-        let extendedDataSet = convertArgs(args: args, completionHandler: completionHandler)
+        let extendedDataSet: [ExtendedData]
+        do {
+            extendedDataSet = try convertArgs(args: args)
+        } catch {
+            completionHandler(nil, createErrorResponse(#function, error))
+            return
+        }
         
-        PodApi.shared.polyIn.deleteQuads(quads: extendedDataSet) { didDelete in
-            if didDelete {
-                completionHandler(MessagePackValue(), nil)
-            } else {
-                completionHandler(nil, MessagePackValue("Failed"))
+        PodApi.shared.polyIn.deleteQuads(quads: extendedDataSet) { error in
+            if let error = error {
+                completionHandler(nil, createErrorResponse(#function, error))
+                return
             }
+            completionHandler(MessagePackValue(), nil)
         }
     }
     
     private func handlePolyInHas(args: [Any], completionHandler: @escaping (MessagePackValue?, MessagePackValue?) -> Void) {
-        let extendedDataSet = convertArgs(args: args, completionHandler: completionHandler)
+        let extendedDataSet: [ExtendedData]
+        do {
+            extendedDataSet = try convertArgs(args: args)
+        } catch {
+            completionHandler(nil, createErrorResponse(#function, error))
+            return
+        }
         
         PodApi.shared.polyIn.hasQuads(quads: extendedDataSet) { doesHave in
             if doesHave {
                 completionHandler(MessagePackValue(), true)
             } else {
-                completionHandler(false, MessagePackValue("Failed"))
+                // TODO: This needs a closer look - should we really fail if the
+                //       data we check for is not there?
+                completionHandler(false, createErrorResponse(#function, "Not there"))
```

Comment:
OK, and now you're using false here while you were using nil everywhere else... Or true. This is probably OK, but I'm kinda confused.

Reply:
Just left it alone since I didn't want to change semantics while improving error messages - but it is very suspicious.